### PR TITLE
[EMB-170] Removing and refurbishing cruft

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
Remove as much custom logic from our ember-data classes as possible, and clean up anything left.

## Summary of Changes
- Remove dirty relationship tracking, automatic updates
  - Use built-in relationship serialization instead of making separate requests
  - Use some ugly private API to preserve the only thing of value: Serializing changed relationships only
- Remove contributor management methods on `user` model
- Refactor embed handling
- Update new project task to create a node with institutions in one request

## Side Effects / Testing Notes
No visible changes, at most would require a regression test.

For future development, updating relationships works differently, since they can now be included in a normal POST/PATCH on an object. It should be a little less magic/opaque now.
- When creating new objects for a relationship, save them separately.
- When adding existing objects to a relationship, just save the one object. e.g. adding institutions to a node just requires `node.save()`

## Ticket

https://openscience.atlassian.net/browse/EMB-170

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: app/adapters/osf-adapter.ts [hunk 1]
```diff
@@ -24,13 +19,14 @@ const { JSONAPIAdapter } = DS;
  * @extends DS.JSONAPIAdapter
  * @uses GenericDataAdapterMixin
  */
-export default class OsfAdapter extends JSONAPIAdapter.extend(GenericDataAdapterMixin, {
+export default class OsfAdapter extends JSONAPIAdapter.extend(GenericDataAdapterMixin).extend({
```
Comment (binoculars) [inline app/adapters/osf-adapter.ts]: Double extend? I haven't seen it before. Maybe add a comment for why?

File: app/serializers/osf-serializer.ts [hunk 2]
```diff
@@ -1,96 +1,110 @@
...
             }
-            if (Array.isArray(data)) {
-                included = included.concat(data);
+
+            // Push the embedded data into the store.  Since it will use this serializer again,
+            // this is indirectly recursive and will extract nested embeds.
+            this.get('store').pushPayload(embeddedObj);
+
+            // Merge links on the embedded object with links on the relationship, so all returned links are available
+            const embeddedLinks = Object.assign(
```
Comment (binoculars) [inline app/serializers/osf-serializer.ts]: You could use the [object spread operator](https://github.com/tc39/proposal-object-rest-spread) here.

File: app/serializers/osf-serializer.ts [hunk 3]
```diff
@@ -1,96 +1,110 @@
...
+                {},
+                embeddedObj.links,
+                resourceHash.relationships[relName].links,
+            );
+
+            // Construct a new relationship in JSON API format
+            if (Array.isArray(embeddedObj.data)) {
+                relationships[relName] = {
+                    data: embeddedObj.data.map(o => ({ id: o.id, type: o.type })),
```
Comment (binoculars) [inline app/serializers/osf-serializer.ts]: Can use destructuring and then take advantage of shorthand property names

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Description

File: app/adapters/osf-adapter.ts (hunk 1 above)
Comment (aaxelb) [reply app/adapters/osf-adapter.ts]: Hm, for some reason I thought that was required for typescript to get the types from the mixin properly, but that's not true. Reverting...

File: app/styles/_variables.scss [hunk 4]
```diff
@@ -34,7 +34,7 @@ $color-bg-gray-light: #efefef;
 $color-bg-gray-blue-dark: #34495e;
 $color-bg-blue-dark: #337ab7;
 $color-bg-blue-light: #def;
-$color-bg-red: #f00;
+$color-bg-red: #a00;
```
[the remark this answers is not in the document]
Comment (aaxelb) [reply app/styles/_variables.scss]: Also, the deeper red is more friendly.

[the remark this answers is not in the document]
Comment (jamescdavis) [reply app/styles/_variables.scss]: TBH #f00 hurt my eyes :)

**Pull request merged**